src/workerd/io/trace-stream.c++

```diff
@@ -1011,39 +1012,18 @@ kj::Promise<WorkerInterface::CustomEvent::Result> TailStreamCustomEventImpl::sen
     capnp::HttpOverCapnpFactory& httpOverCapnpFactory,
     capnp::ByteStreamFactory& byteStreamFactory,
     rpc::EventDispatcher::Client dispatcher) {
-  auto revokePaf = kj::newPromiseAndFulfiller<void>();
-
-  // TODO(streaming-tail): Session is currently being reported as canceled when using
-  // TailStreamCustomEventImpl via RPC, investigate.
-  KJ_DEFER({
-    if (revokePaf.fulfiller->isWaiting()) {
-      revokePaf.fulfiller->reject(KJ_EXCEPTION(DISCONNECTED, "Streaming tail session canceled"));
-    }
-  });
-
   auto req = dispatcher.tailStreamSessionRequest();
   auto sent = req.send();
 
   rpc::TailStreamTarget::Client cap = sent.getTopLevel();
 
-  cap = capnp::membrane(kj::mv(cap), kj::refcounted<RevokerMembrane>(kj::mv(revokePaf.promise)));
```

Comment:
IMO you should keep the revoker as a failsafe to make sure that capabilities cannot leak from the event.

Reply:
With the other changes, this should no longer be causing a problem, right?

Reply:
Odd. It looks like the `WorkerInterface::customEvent()` call is being placed in the global `waitUntilTasks`, so why would it ever be canceled?

Reply:
Resolved in the latest change set (which turns out to be fully disjoint from the last attempt), with help from Dan. Fixing waitUntil lifetimes and making sure that the capability doesn't get deallocated prematurely on the client side resolves all prior test issues.